### PR TITLE
Add thread name in the error message when thread creation fails.

### PR DESCRIPTION
**Pull request by burgerkingeater**
This PR adds thread name in the error message when thread creation fails. It helps with debugging when thread creation fails.

### CODE REVIEW

**Comment by burgerkingeater** [comment]
@hawkinsp would you mind taking a look? Thanks.

**Comment by burgerkingeater** [comment]
@rohan100jain do you know about the windows bazel failure? Why is it happening only on windowns?

File: tensorflow/core/platform/default/env.cc [hunk 1]
```diff
@@ -70,7 +70,7 @@ class PThread : public Thread {
     }
     int ret = pthread_create(&thread_, &attributes, &ThreadFn, params);
     // There is no mechanism for the thread creation API to fail, so we CHECK.
-    CHECK_EQ(ret, 0) << "Thread creation via pthread_create() failed.";
+    CHECK_EQ(ret, 0) << "Thread " + name + " creation via pthread_create() failed.";
```
Comment (allenlavoie) [inline tensorflow/core/platform/default/env.cc:73]: I'd use `<<` instead of `+` here

Comment (burgerkingeater) [reply tensorflow/core/platform/default/env.cc]: thanks, updated.

**Review by allenlavoie — APPROVED** [review]

**Comment by burgerkingeater** [comment]
@allenlavoie thanks for the approval. Please let me know when it can be merged.

**Comment by shuzhang1989** [comment]
@burgerkingeater useful diff, thanks!

**Pull request merged**